saber/boss_access/boss_access.py

```diff
@@ -67,70 +64,42 @@ def boss_pull_cutout(args):
     EXP_NAME = args.exp
     CHAN_NAME = args.chan
 
+    # Use intern to get d_type and i_type
+    intern_chans = array("bossdb://" + COLL_NAME + '/' + EXP_NAME + '/' + CHAN_NAME)
+
     # Create or get a channel to write to
     chan_setup = ChannelResource(
-        CHAN_NAME, COLL_NAME, EXP_NAME, type=args.itype, datatype=args.dtype
+        CHAN_NAME,
+        COLL_NAME,
+        EXP_NAME,
+        type=intern_chans._channel.type,
+        datatype=intern_chans.dtype
```

Comment:
have we tested this code with creating a new channel? I think the array function creates a new channel when it doesnt exist so it might not need the secondary logic